Assets/MixedRealityToolkit/Services/BaseDataProviderAccessCoreSystem.cs

```diff
@@ -19,13 +19,23 @@ public abstract class BaseDataProviderAccessCoreSystem : BaseCoreSystem, IMixedR
     {
         private List<IMixedRealityDataProvider> dataProviders = new List<IMixedRealityDataProvider>();
 
+        protected bool isDestroyed = false;
+
+        /// <inheritdoc />
+        public override void Destroy()
+        {
+            isDestroyed = true;
+            base.Destroy();
+        }
+
         public override void Reset()
         {
             base.Reset();
 
             foreach(var provider in dataProviders)
             {
                 provider.Reset();
+                if (isDestroyed) { break; }
```

Comment:
does this mean that some providers would be reset, but others wouldn't? What would be the consequence of that?

Reply:
good question(s). i will look into that some more

Reply:
this is intentional. as part of the disable/destroy process, the system (ex: input) will have unloaded the data providers that are being iterated